Code/Framework/AzFramework/AzFramework/Spawnable/Script/SpawnableScriptAssetRef.h

```diff
@@ -36,6 +36,16 @@ namespace AzFramework::Scripts
         AZ::Data::Asset<Spawnable> GetAsset() const;
 
     private:
+        class SerializationEvents : public AZ::SerializeContext::IEventHandler
```

Comment:
I have debugged this issue with script canvas. and this is getting called and this bug is not happening. I don't know much beyond that?

Reply:
The reason this works I think is because if any other system updates the spawnable,(for example, saving the base prefab would update the spawnable), this would get triggered and will point to the updated spawnable asset. 

@NicholasVanSickle , I think what you are asking is whether saving SC will trigger this? Saving a SC should cause the spawnable asset to be serialized again. So, I think this won't be hit in that case. Can you explain a particular use case you are concerned about. Maybe @AMZN-mnaumov can confirm if that causes issues with his fix or not